app/views/bot_detection/challenge.html.erb

```diff
@@ -0,0 +1,61 @@
+<% content_for(:head) do %>
+    <script src="<%= cf_turnstile_js_url %>" async defer></script>
+    <meta name="robots" content="noindex">
+  <% end %>
+  
+  
+  <h1 class="brand-alt-h1 mb-4">Traffic control and bot detection...</h2>
+  
+  <div
+    class="cf-turnstile"
+    data-sitekey="<%= cf_turnstile_sitekey %>"
+    data-callback="turnstileCallback"
+  ></div>
+  
+  <noscript>
+    <div class="alert alert-danger">Sorry, Javascript is required to be enabled for our traffic check, and does not appear available.</div>
+  </noscript>
+  
+  <p>We strive to make our content freely available. If this check is preventing you from making use of our resources, make sure you have cookies enabled. If you still have trouble, please <a href="mailto:digital@sciencehistory.org">get in touch</a>. You can also take a look at our information on <a href="https://digital.sciencehistory.org/api_docs">API and Machine Access</a>.</p>
+  
+  <script type="text/javascript">
+    async function turnstileCallback(token) {
+      try {
+        const csrfToken = document.querySelector("[name='csrf-token']");
+  
+        const response = await fetch('<%= "/challenge" %>', {
+          method: 'POST',
+          headers: {
+            "X-CSRF-Token": csrfToken?.content,
+            "Content-Type": "application/json"
+          },
+          body: JSON.stringify({ cf_turnstile_response: token }),
+        });
+  
+        if (!response.ok) {
+          throw new Error('bad response: ' + response.status + ": " + response.url);
+        }
+  
+        result = await response.json();
+        if (result["success"] == true) {
+          const dest = new URLSearchParams(window.location.search).get("dest");
+          window.location.replace("/catalog");
```

Comment:
I think we should still use the `dest` param, incase this is a saved search. Sounds like that's not uncommon

Reply:
Totally. I put this in here *To Just Get It Working* because i wanted it at least working lol. It was doing something funky with the destination Url that it was pulling. It was coming up as nil and I didn't really look into why. I saw it was replacing the subdirectory with whatever was nil. So I just dropped catalog in there.

So yeah, this implementation only works for catalog at the moment.

Do you think its worth while to get this in at least and get it tested and onto prod while I extend it?

Reply:
Sure, if you wanna start getting this tested that's fine